plugins/woocommerce/src/Internal/Logging/RemoteLogger.php

```diff
@@ -84,6 +250,57 @@ private function is_latest_woocommerce_version() {
 		return version_compare( WC()->version, $latest_wc_version, '>=' );
 	}
 
+	/**
+	 * Check if the error exclusively contains third-party stack frames for fatal-errors source context.
+	 *
+	 * @param string $message The error message.
+	 * @param array  $context The error context.
+	 *
+	 * @return bool
+	 */
+	protected function is_third_party_error( string $message, array $context ): bool {
+		// Only check for fatal-errors source context.
+		if ( ! isset( $context['source'] ) || 'fatal-errors' !== $context['source'] ) {
+			return false;
+		}
+
+		// If backtrace is not available, we can't determine if the error is third-party. Log it for further investigation.
+		if ( ! isset( $context['backtrace'] ) || ! is_array( $context['backtrace'] ) ) {
+			return false;
+		}
+
+		$wc_plugin_dir = StringUtil::normalize_local_path_slashes( WC_ABSPATH );
+
+		// Check if the error message contains the WooCommerce plugin directory.
+		if ( is_string( $message ) && str_contains( $message, $wc_plugin_dir ) ) {
```

Comment:
Also feel free to skip (as it shouldn't cause any problems) but, now that we are using strong typing in the signature, we don't need to test if `$message` is a string.

```suggestion
		if ( str_contains( $message, $wc_plugin_dir ) ) {
```